linkml_runtime/utils/schemaview.py

```diff
@@ -1832,24 +1836,28 @@ def usage_index(self) -> dict[ElementName, list[SchemaUsage]]:
 
         :return: dictionary of SchemaUsages keyed by used elements
         """
-        roles = ["domain", "range", "any_of", "exactly_one_of", "none_of", "all_of"]
+        attrs = ["domain", "range", "any_of", "exactly_one_of", "none_of", "all_of"]
         ix = defaultdict(list)
         for cn, c in self.all_classes().items():
             direct_slots = c.slots
             for sn in self.class_slots(cn):
-                s = self.induced_slot(sn, cn)
-                for k in roles:
-                    v = getattr(s, k)
-                    vl = v if isinstance(v, list) else [v]
-                    for x in vl:
-                        if x is not None:
-                            if isinstance(x, AnonymousSlotExpression):
-                                x = x.range
-                                k = f"{k}[range]"
-                            k = k.split("[")[0] + "[range]" if "[range]" in k else k
-                            u = SchemaUsage(used_by=cn, slot=sn, metaslot=k, used=x)
-                            u.inferred = sn in direct_slots
-                            ix[x].append(u)
+                slot = self.induced_slot(sn, cn)
```

Comment:
Renamed these variables whilst I was trying to work out how this method worked. Created a couple of new variables to prevent linter complaints about editing an iterator during looping.